### DISCUSSION TITLE
windows rustdesk huge memory leak

### DISCUSSION THREAD

**Discussion by key2p** [rustdesk/rustdesk#8976] [General]
### Bug Description

RT, rustdesk.exe started a month ago, use 7.0 G memory.

rustdesk version:  1.2.6.44 , download from github release

![image](https://github.com/user-attachments/assets/4080dad8-4302-4a2d-a26c-cffd4a555454)

![image](https://github.com/user-attachments/assets/66ec559c-3e5e-4485-897a-76ed2377fca5)

rustdesk.exe start at a month ago

![image](https://github.com/user-attachments/assets/c093411c-dc47-477e-b8b6-d4f9411750e4)

memory region
![image](https://github.com/user-attachments/assets/1a0a8298-0e05-4d9c-abeb-010e56238d9e)

memory dump (use 7z to extract， github can not upload > 25M )
https://f.ws28.cn/f/es40xnqgpx4 

### How to Reproduce

currently unknown

### Expected Behavior

memory usage normal

### Operating system(s) on local side and remote side

windows 11

### RustDesk Version(s) on local side and remote side

1.2.6.44

### Screenshots

![image](https://github.com/user-attachments/assets/66ec559c-3e5e-4485-897a-76ed2377fca5)

**Comment by 21pages** [comment]
Thanks for your feedback, we need some information to reproduce.
1. It's installed or portable? Usually used as control or cotrolled side? Which codec is used usually?
2. What's the value of these two options on that client?
![1722995726452](https://github.com/user-attachments/assets/a2f862e8-8e1f-4ea8-86bb-424cacab9dd5)

**Reply by key2p** [reply]
1. installed. run as  server mode(controlled side).  
the process command line is : xxxx\rustdesk.exe -server

3. use default config, none manually or special config. 
4. use self host server.
5. hardware: intel 12700,

![image](https://github.com/user-attachments/assets/f74a1d22-90fb-4a37-9f9d-b0ed740b0875)
![image](https://github.com/user-attachments/assets/12c640bb-9825-4452-a796-f40db1c1818a)

**Reply by rustdesk** [reply]
If possible, please observe more carefully, you may find some pattern or scenario when the memory leak starts and accumulate.

**Reply by rustdesk** [reply]
Could you try different codec?

![image](https://github.com/user-attachments/assets/4a47823e-e528-42ec-a515-801fe71adb12)

**Comment by xlionjuan** [comment]
I have observed this for a Windows computer that didn't reboot for very long time+never close RustDesk windows

Just close all RustDesk windows resolved this issue (service won't stop if you close windows, don't worry)

**Reply by rustdesk** [reply]
Still can easy reproduce now? e.g. run 1 hour, and see the memory change.

**Reply by xlionjuan** [reply]
It will growing very slowly, it will become noticeable after one or two weeks.

(Only window will memeoy leak, service won't, so won't bother if it is only being controlled, or user will closed the window if they didn't use.)

I can still reproduced on 1.3.0.

**Reply by rustdesk** [reply]
@21pages open your window for one week. It might be related to the IPC or somewhere else. Also open sciter version on the same machine if possible, othewise open on another machine.

**Reply by rustdesk** [reply]
@xlionjuan only on one machine or all your machines? Which version start to have this issue?

**Reply by xlionjuan** [reply]
It is a computer in my school, both incoming/outgoing connection are frequently, so this is why its window is opened and never closed 

For version, I can't sure.

**Reply by rustdesk** [reply]
> both incoming/outgoing connection are frequently

@21pages  note this. One week is noticeable, I guess one hour is enough to see something.

**Comment by maomaochong199** [comment]
一样的问题，1.2.7版本，安装版，待机状态就会一直占内存，![这是图片](https://shp.qpic.cn/collector/446286604/cc4dcbad-e0ca-4aa1-aa12-00cb1d6c5d65/0/895614891498.png) @rustdesk

**Reply by rustdesk** [reply]
The service or GUI?

**Reply by maomaochong199** [reply]
win x64 gui ，换到1.3.1的nightly版使用了一会儿，好像没有再涨到200m以上了，是修过了么，已切换到1.3.1使用

**Reply by rustdesk** [reply]
Yes, we had a fix in 1.3.0 for this, https://github.com/rustdesk/rustdesk/issues/8899 but for osx, may be related, please keep watch.

**Reply by maomaochong199** [reply]
只要开着这个窗口，内存还是会缓慢变大，一晚上长到700m了，![f27dd2fee004eb2ea480d20d0d95915e](https://github.com/user-attachments/assets/4c7b7d40-b657-4897-8968-5f4242000ae0)

**Reply by rustdesk** [reply]
@21pages

**Reply by rustdesk** [reply]
I reproduced on my mac, but not on windows. the memory increased from 60m to 190m in 12 hours.

**Reply by mircboe** [reply]
Windows 10
![image](https://github.com/user-attachments/assets/5c730958-e437-4418-bf49-86d537707164)
![image](https://github.com/user-attachments/assets/1f4871dc-d3ed-4448-8859-5e7b3d2f2377)

**Reply by 21pages** [reply]
What's the version, 1.3.5 release or nightly or old version? controll or controlled side?

**Reply by 21pages** [reply]
> Windows 10
> ![image](https://github.com/user-attachments/assets/5c730958-e437-4418-bf49-86d537707164)
> ![image](https://github.com/user-attachments/assets/1f4871dc-d3ed-4448-8859-5e7b3d2f2377)
> 

What's the command line of the huge memory process? Installed or portable?
